src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -568,13 +569,23 @@ public bool ApplyNullableTransforms(ImmutableArray<bool> transforms, ref int pos
                 result = result.DoUpdate(newTypeSymbol, result.CustomModifiers);
             }
 
-            if (isNullable)
+            //if (!result.IsValueType) // PROTOTYPE(NullableReferenceTypes): this is causing cycle 6 in NullableAttribute_01
```

Comment:
>// PROTOTYPE(NullableReferenceTypes): this is causing cycle 6 in NullableAttribute_01 [](start = 39, length = 85)

Does this cycle still occur?

Reply:
Yes, there is still a cycle:

```C#
TypeSymbolWithAnnotations.IsValueType.get() Line 333
TypeSymbolWithAnnotations.ApplyNullableTransforms(ImmutableArray<bool> transforms, bool useNonNullTypes, ref int position, out .TypeSymbolWithAnnotations result) Line 572
NamedTypeSymbol.ApplyNullableTransforms(ImmutableArray<bool> transforms, bool useNonNullTypes, ref int position, out .TypeSymbol result) Line 795
TypeSymbolWithAnnotations.ApplyNullableTransforms(ImmutableArray<bool> transforms, bool useNonNullTypes, ref int position, out .TypeSymbolWithAnnotations result) Line 562
Metadata.PE.NullableTypeDecoder.TransformType(.TypeSymbolWithAnnotations metadataType, ImmutableArray<bool> nullableTransformFlags) Line 34
Metadata.PE.NullableTypeDecoder.TransformType(.TypeSymbolWithAnnotations metadataType, System.Reflection.Metadata.EntityHandle targetSymbolToken, .Metadata.PE.PEModuleSymbol containingModule) Line 26
Metadata.PE.PENamedTypeSymbol.DecodeBaseAndInterfaceNullableAndTupleTypes(.NamedTypeSymbol type, System.Reflection.Metadata.EntityHandle handle, .Metadata.PE.PEModuleSymbol moduleSymbol) Line 518
Metadata.PE.PENamedTypeSymbol.MakeDeclaredBaseType() Line 460
Metadata.PE.PENamedTypeSymbol.GetDeclaredBaseType(Roslyn.Utilities.ConsList<Microsoft.CodeAnalysis.CSharp.Symbol> basesBeingResolved) Line 431
Metadata.PE.PENamedTypeSymbol.TypeKind.get() Line 1631
NamedTypeSymbol.IsValueType.get() Line 393
TypeSymbolWithAnnotations.IsValueType.get() Line 333
TypeSymbolWithAnnotations.ApplyNullableTransforms(ImmutableArray<bool> transforms, bool useNonNullTypes, ref int position, out .TypeSymbolWithAnnotations result) Line 572

```

---
In reply to: [198235913](https://github.com/dotnet/roslyn/pull/27807#discussion_r198235913) [](ancestors = 198235913)

Reply:
Thanks. It should be addressed in #28028.

---
In reply to: [198272037](https://github.com/dotnet/roslyn/pull/27807#discussion_r198272037) [](ancestors = 198272037,198235913)